scripts/build/Jenkins/Jenkinsfile

```diff
@@ -336,51 +416,281 @@ def PreBuildCommonSteps(Map pipelineConfig, String snapshot, String repositoryNa
+                        }
+                        if(templatesExist) {
+                            PlatformSh("scripts/o3de${ext} register --all-templates-path ${workspace}/${repo.REPOSITORY_NAME}/Templates || ver>nul", "Registering all templates in ${workspace}/${repo.REPOSITORY_NAME}/Templates")
+                        }
+                        if(restrictedExist) {
+                            PlatformSh("scripts/o3de${ext} register --all-restricted-path ${workspace}/${repo.REPOSITORY_NAME}/Restricted || ver>nul", "Registering all restricted in ${workspace}/${repo.REPOSITORY_NAME}/Restricted")
+                        }
+                        //enable all gems in this repo
+                        if(gemsExist) {
+                            PlatformSh("scripts/o3de${ext} enable-gem --all-gem-paths ${workspace}/${repo.REPOSITORY_NAME}/Gems --project-name AutomatedTesting || ver>nul", "Enabling all gems in AutomatedTesting")
+                        }
+
+                        //if its a singular repo register it using the --<type>-path
+                        if(engineRepo) {
+                            PlatformSh("scripts/o3de${ext} register --engine-path ${workspace}/${repo.REPOSITORY_NAME} || ver>nul", "Registering engine in ${workspace}/${repo.REPOSITORY_NAME}")
+                        }
+                        if(projectRepo) {
+                            PlatformSh("scripts/o3de${ext} register --project-path ${workspace}/${repo.REPOSITORY_NAME} || ver>nul", "Registering project in ${workspace}/${repo.REPOSITORY_NAME}")
+                        }
+                        if(gemRepo) {
+                            PlatformSh("scripts/o3de${ext} register --gem-path ${workspace}/${repo.REPOSITORY_NAME} || ver>nul", "Registering gem in ${workspace}/${repo.REPOSITORY_NAME}")
+                        }
+                        if(templateRepo) {
+                            PlatformSh("scripts/o3de${ext} register --template-path ${workspace}/${repo.REPOSITORY_NAME} || ver>nul", "Registering template in ${workspace}/${repo.REPOSITORY_NAME}")
+                        }
+                        if(restrictedRepo) {
+                            PlatformSh("scripts/o3de${ext} register --restricted-path ${workspace}/${repo.REPOSITORY_NAME} || ver>nul", "Registering restricted in ${workspace}/${repo.REPOSITORY_NAME}")
+                        }
+                        //if this singular repo is a gem enable it
+                        if(gemRepo) {
+                            PlatformSh("scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repo.REPOSITORY_NAME} --project-name AutomatedTesting || ver>nul", "Enabling ${gemName} in AutomatedTesting")
+                        }
+                    }
+
+                    //build
+                    PlatformSh(command, "Running ${platform} ${type}")
```

Comment:
Shouldn't the build command also be located outside the external repo loop. Otherwise it won't run if there are no external repos defined. 

I noticed in the branch pipeline the register steps ran, but not the build step.

Reply:
moved this to the engine scope